src/FireHydrantPhysicsComponent.cpp

```diff
@@ -0,0 +1,60 @@
+#include "FireHydrantPhysicsComponent.h"
+
+#include "GameManager.h"
+#include "GameObject.h"
+#include <glm/gtx/rotate_vector.hpp>
+
+FireHydrantPhysicsComponent::FireHydrantPhysicsComponent() {}
+
+FireHydrantPhysicsComponent::~FireHydrantPhysicsComponent() {}
+
+void FireHydrantPhysicsComponent::initObjectPhysics() {
+   gravity = 10.0f;
+   yVelocity = 0.0f;
+}
+
+void FireHydrantPhysicsComponent::updatePhysics(float deltaTime) {
+   GameWorld& world = GameManager::instance().getGameWorld();
+   std::shared_ptr<GameObject> objHit = world.checkCollision(holder_);
+
+   glm::vec3 oldPosition = holder_->getPosition();
+   glm::vec3 newPosition = holder_->getPosition() + (holder_->velocity *
+      holder_->direction * deltaTime);
+
+   newPosition += glm::vec3(0.0, yVelocity * deltaTime, 0.0);
+   holder_->setPosition(newPosition);
+   updateBoundingBox();
+
+   if (holder_->getPosition().y > 1.0f) {
+      yVelocity -= gravity * deltaTime;
+   } else {
+      yVelocity = 0.0f;
+      holder_->velocity = 0.0f;
+   }
+
+   if (objHit != nullptr) {
+      GameObjectType objTypeHit = objHit->type;
+      
+      if (objTypeHit == GameObjectType::PLAYER) {
+         glm::vec3 reactDir = objHit->direction;
+         glm::vec3 rotAxis = glm::cross(objHit->direction, glm::vec3(0, 1, 0));
+         // Give the direction of the hydrant a vertical component.
+         reactDir = glm::rotate(reactDir, (float)(M_PI / 6.0f), rotAxis);
```

Comment:
We should definitely introduce a random component to the velocity. After we update the matrix math a rotation while it's in the air. (Same for the cookie btw)